src/ty.rs

```diff
@@ -984,7 +998,21 @@ mod printing {
             self.turbofish.to_tokens(tokens);
             self.lt_token.to_tokens(tokens);
             self.lifetimes.to_tokens(tokens);
+            if !self.lifetimes.empty_or_trailing() && !self.types.is_empty() {
+                tokens::Comma::default().to_tokens(tokens);
+            }
             self.types.to_tokens(tokens);
+            if (
```

Comment:
This conditional is the one which I gave up the most on :-/. Not sure how to make it not-ugly.